internal/server/handlers_test.go

```diff
@@ -219,3 +219,87 @@ func TestPseudoDeltaSotWHandler(t *testing.T) {
 	require.WithinDuration(t, time.Now(), start.Add(wait), 10*time.Millisecond)
 	require.ElementsMatch(t, []*anypb.Any{testutils.MustMarshal(t, bazR).Resource}, lastRes.Resources)
 }
+
+func TestHandlerBatchingWithIRV(t *testing.T) {
+	var released atomic.Bool
+	ch := make(chan map[string]*ads.RawResource)
+	handler := newHandler(
+		testutils.Context(t),
+		NoopLimiter{},
+		NoopLimiter{},
+		new(customStatsHandler),
+		false,
+		func(resources map[string]*ads.RawResource) error {
+			ch <- maps.Clone(resources)
+			return nil
+		},
+	)
+	notify := func(name string, resource *ads.RawResource) {
+		handler.Notify(name, resource, ads.SubscriptionMetadata{})
+	}
+
+	expectedEntries := make(map[string]*ads.RawResource)
```

Comment:
Can you avoid sharing state between tests like this? You can either declare `expectedEntries` inside each test, or better yet replace your `require` check with this:
```
		require.Equal(t, map[string]*ads.RawResource{barResource.Name: barResource}, <-ch)
```